src/Protocols/Wishbone/Standard/Hedgehog.hs

```diff
@@ -239,6 +338,67 @@ driveStandard ExpectOptions {..} reqs =
         =
         transferToSignals d : (rep `C.seqX` go 0 (d : dat) replies)
 
+-- | Circuit which validates the wishbone communication signals between a
+--   master and a slave circuit.
+--
+--   Halts execution using 'error' when a "common sense" spec validation occurs.
+--
+--   N.B. Not synthesisable.
+validatorCircuit ::
+  forall dom addressWidth a.
+  ( HasCallStack,
+    HiddenClockResetEnable dom,
+    KnownNat addressWidth,
+    BitPack a,
+    NFDataX a,
+    ShowX a
+  ) =>
+  Circuit (Wishbone dom 'Standard addressWidth a) (Wishbone dom 'Standard addressWidth a)
+validatorCircuit =
+  Circuit $ mealyB go (0 :: Integer, (emptyWishboneM2S, emptyWishboneS2M), CSVSQuiet)
+ where
+  go (cycle, (m2s0, s2m0), state0) (m2s1, s2m1) =
+    case nextStateCommonSense state0 m2s0 s2m0 of
+      Left err -> error $
+        "Wishbone common-sense validation error on cycle "
+        <> show cycle
+        <> ": "
+        <> err
+        <> "\n"
+        <> showX m2s0
+        <> "\n"
+        <> showX s2m0
```

Comment:
Ah! That's nice, thanks! I realised I don't need to use `showX` here either since `Show (Wishbone(S2M|M2S))` is implemented via `ShowX` too! I'll change this

Reply:
Reverted as this would pull in all of `clash-lib`

Reply:
sorry!